### ISSUE TITLE
Better UX for `SHOW SOURCES` on Postgres Sources

### ISSUE THREAD

**Issue by nmeagan11** [MaterializeInc/materialize#15515]
### Feature request

When running `SHOW SOURCES;` for a Postgres source where the `SIZE` is not `3xsmall`, the user will see each subsource as `3xsmall`, which is confusing. For example: 

```
SHOW SOURCES;

         name         |   type    |  size
----------------------+-----------+---------
 table_1              | subsource | 3xsmall
 table_2              | subsource | 3xsmall
 ...                  | subsource | 3xsmall
 pg_source            | postgres  | large
```

We should think about how to better represent this output.

**Comment by nmeagan11** [comment]
cc @ggnall

**Comment by sjwiesman** [comment]
In my mind, size should be n/a or match the size of the parent source

**Comment by benesch** [comment]
I think `NULL` is best because it does not correspond to any actual resources!

**Comment by benesch** [comment]
Note to self: what happens if you `ALTER SIZE` on a subsource? 🤦🏽‍♂️

**Comment by nmeagan11** [comment]
Linking a related documentation PR: https://github.com/MaterializeInc/materialize/pull/15370#issue-1406819127

**Comment by benesch** [comment]
Well, good news! This already got fixed in #15222. Not sure if accidental or intentional, but definitely desired—so thanks, @sploiselle!
